types/src/aggregator.rs

```diff
@@ -83,7 +83,9 @@ impl TryIntoMoveValue for DelayedFieldID {
         Ok(match layout {
             MoveTypeLayout::U64 => Value::u64(self.as_u64()),
             MoveTypeLayout::U128 => Value::u128(self.as_u64() as u128),
-            layout if is_string_layout(layout) => bytes_to_string(to_utf8_bytes(self.as_u64())),
+            layout if is_string_layout(layout) => {
+                bytes_to_string(u64_to_fixed_size_utf8_bytes(self.as_u64()))
```

Comment:
can you add a comment here that we need to have a fixed sized identifier representation, to have consistent and deterministic gas charging?